test/javascripts/acceptance/plugin-outlet-decorator-test.js.es6

```diff
@@ -0,0 +1,64 @@
+import { acceptance } from "helpers/qunit-helpers";
+import { withPluginApi } from "discourse/lib/plugin-api";
+import { resetDecorators } from "discourse/components/plugin-connector";
+
+const PREFIX = "javascripts/single-test/connectors";
+acceptance("Plugin Outlet - Decorator", {
+  loggedIn: true,
+
+  beforeEach() {
+    Ember.TEMPLATES[
+      `${PREFIX}/discovery-list-container-top/foo`
+    ] = Ember.HTMLBars.compile("FOO");
+    Ember.TEMPLATES[
+      `${PREFIX}/discovery-list-container-top/bar`
+    ] = Ember.HTMLBars.compile("BAR");
+
+    withPluginApi("0.8.38", api => {
+      api.decoratePluginConnector(
+        "discovery-list-container-top",
+        (elem, args) => {
+          if (elem.classList.contains("foo")) {
+            elem.style.backgroundColor = "yellow";
+
+            if (args.category) {
+              elem.classList.add("in-category");
+            } else {
+              elem.classList.remove("in-category");
+            }
+          }
+        },
+        { id: "yellow-decorator" }
+      );
+    });
+  },
+
+  afterEach() {
+    delete Ember.TEMPLATES[`${PREFIX}/discovery-list-container-top/foo`];
+    delete Ember.TEMPLATES[`${PREFIX}/discovery-list-container-top/bar`];
+
+    resetDecorators();
```

Comment:
This should probably be done in the test helpers. Otherwise if people have acceptance tests for their plugins they could leak.